aesara/graph/fg.py

```diff
@@ -484,7 +484,9 @@ def replace(
                 f"rewriting: rewrite {reason} replaces {var} of {var.owner} with {new_var} of {new_var.owner}"
             )
 
+        name = new_var.name
         new_var = var.type.filter_variable(new_var, allow_convert=True)
+        new_var.name = name
```

Comment:
Do we need this in general? In PyMC I added a fix here: https://github.com/pymc-devs/pymc/pull/6336/commits/dc3323a4fb9db09863b059160fcfb31ed4b0fc1c

I am not sure we want to call `SpecifyShape(x)`, `x.name` in general. We might need to, but what was the issue you found?

Reply:
Also in some rewrites like MergeOptimizer, you may be erasing the name if the "reference variable" name is `None` or something else

Reply:
I think that was to fix the same issue you fixed here: https://github.com/pymc-devs/pymc/pull/6336/commits/3118cc646c6b7a4324184274fdbf545a5a32773b
I'll remove the commit in the rebase.

Reply:
done